doc/reverse_engineering/game_mechanics/market.md

```diff
@@ -1,28 +1,32 @@
 # Market
 
-[Source](https://www.youtube.com/watch?v=NvsM9B4ac2g)
+Source: [here](http://ageofempires.wikia.com/wiki/Market_(Age_of_Empires_II)) for the first part - verified through reverse engeneering.
+
+[here](https://www.youtube.com/watch?v=NvsM9B4ac2g) for the rest.
 
 Gold from trading calculation, tribute system, buying/selling resources and the destruction of markets.
 
-## Distance Gold Calculation
+## Trade Cart gold calculation
 
 The gold you receive from trading with a market depends on the *distance in tiles* (or *distance in tile width*) between the two markets. The further the markets are apart from each other, the more gold is generated by a trade cart.
 
-Generated amount of gold is not linear to the distance in tiles. Instead the algorithm probably uses a 2nd degree function (see below in Section **Math**) to calculate the trade value. Since the market uses 4x4 tiles, the measuring point seems to be in the center of the 4x4 tiling.
-
-Note that the gold value shown when you select the market is rounded but the gold you receive is not. This means that in the case of two markets being placed next to each other, trade carts will still generate a tiny fraction of gold.
+Generated amount of gold is not linear to the distance in tiles. Note that the gold value shown when you select the market is rounded but the gold you receive is not. This means that in the case of two markets being placed next to each other, trade carts will still generate a tiny fraction of gold.
 
 ### Math
 
-    (gold value) = 0.001 * (distance in tiles)^2 + 0.1353 * (distance in tiles)
+    (gold value) = 0.46 * d * (d/size + 0.3)
+
+with
+`size`, the map size in tiles and
+`d` is calculated as follows:
 
-or
+    d = max(0.1, sqrt(max(0, abs(x2-x1)-5)^2 + max(0, abs(x2-x1)-5)^2)
 
-    (gold value) = 0.001 * (distance in tiles)^2 + e^(-2) * (distance in tiles)
+The coefficient `0.46` is calculated as
 
-### Diagonal Calculation
+    2 * 0.345 /(a + 0.5)
 
-AoE2 also takes into account that crossing a tile diagonally takes up to 41 % (for fully diagonal movement) longer. Hence it probably uses trigonometry to calculate the real *distance in tile width* between the two markets.
+where `0.345` is related to the Trade Cart speed and varies if the player has a Spanish ally (normally set to `1`)
```

Comment:
Do you mean that `a` is set to `1` by default?

Reply:
yes, I'll rephrase